src/main/java/com/app/katchup/DistributedCalendar/CalendarController.java

```diff
@@ -25,53 +28,46 @@
 public class CalendarController {
 
     private static final Logger logger = LogManager.getLogger(CalendarController.class);
-
     @Autowired
     CalendarService calendarService;
     @Autowired
     UserService userService;
     @Autowired
     MeetingService meetingService;
-
+    MeetingResponseRepository meetingResponseRepository;
 
     @GetMapping("/{userName}")
     public ResponseEntity<DistributedCalendar> showUserEvents(@PathVariable String userName, HttpServletRequest request) {
-
-        User user = userService.getUserByUserName(userName);
-        if (user == null) {
-            logger.info("user name is not valid");
+        if (!(userService.isCredentialsMatched(userName, request.getHeader("password"))))
             return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
-        }
 
-        boolean t = userService.isCredentialsMatched(user.getUserName(), request.getHeader("password"));
-        if (t) {
-            logger.info("user password not valid");
-            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
-        } else {
-            List<String> meetingIds = calendarService.getAcceptedMeetingIds(user.getUserName());
+        else {
+            List<MeetingID> meetingIds = calendarService.getAcceptedMeetingIds(userName);
+            List<Meeting> meetingList = new ArrayList<>();
 
-            List<Meeting> meetingList = meetingService.getMeetingDetailsForMeetingIds(meetingIds);
+            if (meetingIds.size() <= 0)
+                throw new UnAuthorizedException("Sorry! User is not allowed to enter this meeting");
 
-            DistributedCalendar distributedCalendar = new DistributedCalendar();
+            else if (meetingIds.size() > 0) {
```

Comment:
this else you can remove. because the previous if conditon will filter == 0
so anything that comes here will be >0

Reply:
i am adding if (meetingIds.size() >= 0) and returning an empty event list if no meetings were accepted by the user

Reply:
if no meetings were accepted by the user, then it is == 0 not >= 0. Also the previous if condition checks for <= 0 then why again the same >= is used?

Reply:
removed other if() and now there is only one if(>=0) condition, it returns empty object for no accepted meetings and meeting details for accepted ones with HTTP OK.

Reply:
Okay. If it is empty then check with == 0 and return [] 
 
If not, it means(indirectly >0) then go ahead and get the details and return here

Reply:
done, added the change and pushed to my branch.

Reply:
please check and merge the pull request